File: fundamental-data-structures/array/questions/stocksell/stocksell.go
package questions

// StockSell return max profit
// - if using []int as parameter, then we can't pass array to this function
// - if using [5]int as parameter, then we have limitation here (type-strong)
func StockSell(price []int) int {
	result := 0
	len := len(price)
	if len > 0 {
		for i := 0; i < len-1; i++ {
			gap := price[i+1] - price[i]
			if gap > 0 {
				result += gap
			}
		}

		return result
	}
	return 0
}

File: fundamental-data-structures/array/questions/stocksell/StockSell.kt
package array.questions.stocksell

class StockSellKt {

    fun maxProfit(prices: IntArray): Int {

        // protect from empty
        val len = prices.size
        if (len <= 0) return 0

        // start
        var result = 0
        for (i in 0 until len - 1) {
            val gap = prices[i + 1] - prices[i]
            if (gap > 0) {
                result += gap
            }
        }

        return result
    }
}